#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.052029377755420034 0.8222259984157274 0.5667780441933543 3.2922651916457273
  position 0.3372731824810797 3.0810743442887634 -1.1647050100221699
}
DirectionalLight {
  ambientIntensity 1
  direction 0 -1 0
}
DirectionalLight {
  direction 1 -1 0
}
DirectionalLight {
  direction -1 -1 0
}
DirectionalLight {
  direction 0 -1 -1
}
DirectionalLight {
  direction 0 -1 1
}
RectangleArena {
  translation 0.12 0 -0.28
  floorSize 3 3
  floorTileSize 0.25 0.25
  floorAppearance Parquetry {
    type "dark strip"
    colorOverride 0 0 0
  }
}
Solid {
  translation -0.4 0 -1.63
  scale 0.001 0.001 0.001
  children [
    Shape {
      appearance PBRAppearance {
        roughness 1
        metalness 0
      }
      geometry Mesh {
        url [
          "first_part.STL"
        ]
      }
    }
  ]
  locked TRUE
}
Robot {
  translation -0.275 0.04 0.95
  rotation 0 1 0 3.14159
  children [
    DEF body Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.06 0 0.05
        rotation 0 0 1 1.569999999999998
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        boundingObject USE wheel
        physics DEF physics_wb Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0.05
        rotation 0 0 1 1.5699999999999987
        children [
          USE wheel
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics USE physics_wb
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.06 0 -0.05
        rotation 0 0 1 1.5699999999999967
        children [
          USE wheel
        ]
        name "solid(2)"
        boundingObject USE wheel
        physics USE physics_wb
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.06 0 -0.05
        rotation 0 0 1 1.57
        children [
          USE wheel
        ]
        name "solid(3)"
        boundingObject USE wheel
        physics USE physics_wb
      }
    }
    DistanceSensor {
      translation 0.04 -0.015 0.1
      rotation 0 0 1 -1.57
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "left_1"
      boundingObject USE sensor
      physics Physics {
      }
      type "infra-red"
    }
    DistanceSensor {
      translation 0.02 -0.015 0.1
      rotation 0 0 1 -1.57
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "left_2"
      boundingObject USE sensor
      physics Physics {
      }
      type "infra-red"
    }
    DistanceSensor {
      translation -0.04 -0.015 0.1
      rotation 0 0 1 -1.57
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "right_1"
      boundingObject USE sensor
      physics Physics {
      }
      type "infra-red"
    }
    DistanceSensor {
      translation -0.02 -0.015 0.1
      rotation 0 0 1 -1.57
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "right_2"
      boundingObject USE sensor
      physics Physics {
      }
      type "infra-red"
    }
  ]
  boundingObject USE body
  physics Physics {
  }
  controller "4_sensor_PID"
}
